src/rpc/rawtransaction.cpp

```diff
@@ -1839,115 +1839,91 @@ UniValue sendrawtransaction(const UniValue& params, bool fHelp)
 
     // parse hex string from parameter
     CTransaction tx;
-    if (!DecodeHexTx(tx, params[0].get_str()))
-        throw JSONRPCError(RPC_DESERIALIZATION_ERROR, "TX decode failed");
-    uint256 hashTx = tx.GetHash();
+    // parse hex string from parameter
+    CScCertificate cert;
+    bool isTransaction = true;
+
+    if (!DecodeHexTx(tx, params[0].get_str())) {
+        if (DecodeHexCert(cert, params[0].get_str()))
```

Comment:
We may parse version first, then choose what try to parse: Tx or Cert. It will prevent wasting of time for certificate and/or invalid Txs.
PS: act similar to `signrawtransaction` approach.

Reply:
Added support of prevtxs and sighashtype to the certificate version.